### DISCUSSION TITLE
ROG Ally - Can GHelper change the VRAM allocation same as Armoury Crate?

### DISCUSSION THREAD

**Discussion by trias702** [seerge/g-helper#3435] [General]
Hello,

I'm currently using GHelper with the ROG Ally X and it works great, but wanted to ask if it is possible for GHelper to somehow set the VRAM allocation in the same was as Armoury Crate is able to? I know that you can set this in the BIOS of the device, unfortunately, the ROG Ally X currently has a bug in its latest version 308 BIOS which resets the VRAM allocation to "Auto" on every reboot, and the only way to make a different allocation (e.g. 6GB or 8GB) permanently stick is to use the VRAM allocation in Armoury Crate. However, I don't use AC at all, only GHelper, so am wondering if it is somehow possible to set this with GHelper?

**Comment by seerge** [comment]
@trias702 hello, it can if endpoint responsible for VRAM allocation is same as on my ROG Ally (non X). 

If it’s there - you should see a dropdown for video memory under `Extra` settings. If you don’t see it - it means Asus has changed endpoint, and I need help with discovering it from someone with an X device

**Reply by trias702** [reply]
@seerge Thank you kindly for the response. It does indeed appear that Asus changed the endpoint for the Ally X as I do not see any dropdown for video memory under the `Extra` settings.

That being said, I have an Ally X and would be happy to help you discover the new endpoint. Please let me know what you would like me to do, I'm happy to compile/execute and run any C#/.Net code on the device.

**Reply by seerge** [reply]
@trias702 hello,

It's possible to capture / debug of what AC on Ally X sends to the device in order to change VRAM setting.

Here is the guide on what needs to be set up
https://github.com/seerge/g-helper/wiki/_ACPI-Monitoring-and-Debugging

With the only exception that in Capturing -> Step 1 you will need to record what happens when you change VRAM to different values in AC

Let me know if you want to do that.

Thanks!

**Reply by trias702** [reply]
Thank you kindly for the clear instructions! I will execute these steps sometime next week or the week after and report back my findings.

**Reply by trias702** [reply]
I installed Armoury Crate on my Ally X in preparation of capturing the endpoint, however, I noticed that AC is also currently broken for VRAM on the Ally X, in that changing VRAM in AC requires a restart, and when the device boots up, the VRAM is set correctly. However, any subsequent reboots reset the VRAM back to Auto, which is exactly what happens if you set it manually in the BIOS. Therefore, this seems to be a bug in the current 308 BIOS for the Ally X, and changing it via the endpoint won't matter because it won't persist after the next next reboot anyway.

This is a bug in the BIOS which ASUS needs to fix. I note that this BIOS bug does not exist in the current BIOS of the OG Ally, only in the Ally X.

It would still be nice to discover the Ally X endpoint, but it's not really necessary since we can change the VRAM in the BIOS. But it's all moot until they fix the BIOS. Feel free to mark this question closed.

**Reply by seerge** [reply]
@trias702 ok, then it's probably better to wait for Asus to fix it in BIOS :)) otherwise people will just falsely accuse G-Helper in doing something wrong
